### PR TITLE
ocaml: support for ARM

### PR DESCRIPTION
**Pull request by fxcoudert**
Use upstream patch at https://github.com/ocaml/ocaml/pull/10026

### CODE REVIEW

File: Formula/ocaml.rb [hunk 1]
```diff
@@ -38,6 +38,15 @@ class Ocaml < Formula
     satisfy { HOMEBREW_PREFIX.to_s == "/usr/local" }
   end
```
Comment (aimileus) [inline Formula/ocaml.rb:39]: I think this needs to accept `/opt/homebrew` as prefix on ARM.

Comment (fxcoudert) [reply Formula/ocaml.rb:39]: I know… but it's not building anyway :(

**Comment by aimileus** [comment]
I cherry-picked this mentioned patch to the 4.10.0 tag as https://github.com/aimileus/ocaml/commit/27fd20a so that the patch applies cleanly. I have not extensively tested the compiler, but the Homebrew testcases of both `ocaml` and `coq` succeed. The only conflicts for the commit were the changelog (where I removed all changes except the 'back port to Apple Silicon) and the `configure` script, which I regenerated using `./autogen`.

On the other hand, there is an upstream minor release 4.10.2 that also seems to work. Maybe it is cleaner to just bump this minor version.

**Comment by andreypopp** [comment]
Created #71605 for 4.10.2

**Comment by carlocab** [comment]
🍾 for ARM in #71605.

**Comment by carlocab** [comment]
Bottle jobs dispatched for:

bibtex2html
camlp5
dune
ocamlbuild
ocamlsdl
ocaml-findlib
one-ml
opam
ott
pdfsandwich
unison

**Pull request closed without merging**